src/webviews/extension-side/dataviewer/interpreterDataViewerDependencyImplementation.node.ts

```diff
@@ -0,0 +1,124 @@
+import { IInstaller, Product, InstallerResponse } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { Cancellation, createPromiseFromCancellation } from '../../../platform/common/cancellation';
+import { traceWarning } from '../../../platform/logging';
+import { IPythonExecutionFactory } from '../../../platform/common/process/types.node';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { IInterpreterService } from '../../../platform/interpreter/contracts';
+import { PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { pandasMinimumVersionSupported } from './constants';
+
+/**
+ * Uses the Python interpreter to manage dependencies of a Data Viewer.
+ */
+export class InterpreterDataViewerDependencyImplementation implements IDataViewerDependencyService {
+    constructor(
+        private readonly applicationShell: IApplicationShell,
+        private readonly installer: IInstaller,
+        private pythonFactory: IPythonExecutionFactory,
+        private interpreterService: IInterpreterService,
+        private isCodeSpace: boolean
+    ) {}
+
+    public async checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
+        const tokenSource = new CancellationTokenSource();
+        try {
+            const pandasVersion = await this.getVersion(interpreter, tokenSource.token);
+
+            if (Cancellation.isCanceled(tokenSource.token)) {
+                return;
+            }
+
+            if (pandasVersion) {
+                if (pandasVersion.compare(pandasMinimumVersionSupported) > 0) {
+                    return;
+                }
+                sendTelemetryEvent(Telemetry.PandasTooOld);
+                // Warn user that we cannot start because pandas is too old.
+                const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+                throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
+            }
+
+            sendTelemetryEvent(Telemetry.PandasNotInstalled);
+            await this.installMissingDependencies(interpreter, tokenSource);
+        } finally {
+            tokenSource.dispose();
+        }
+    }
+
+    private async installMissingDependencies(
+        interpreter: PythonEnvironment,
+        tokenSource: CancellationTokenSource
+    ): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!,
+            pythonEnvType: interpreter?.envType
+        });
+        const selection = this.isCodeSpace
```

Comment:
Woudln't it be better to have the code to compare the version and display messages and the like to be in a base class (or seprate class), and have separate implementations for getting version and installing the pacakges.
Else we have to copy this code for comparing versions and displaying messages into two or three places (one for kernel, one for interpreter, one for debugger).

Reply:
Debugger dependency checker (since it checks for an ipykernel version) will be super similar to this so that could be used as a guide to see how to abstract out the commonality.

Reply:
How does this look? https://github.com/microsoft/vscode-jupyter/pull/10825/commits/60dcbf9c6785a8e5841f58bc4b2afb5581491b3d as a first step. Can grow over time 🙂